docs/content/standards/kiosk-apps/nft-rental.mdx

```diff
@@ -0,0 +1,428 @@
+
+```move
+struct Rented has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Listed` struct represents a listed item. The only field the struct includes is the ID of the object. It is used as the DF key in the renter's `Bag <key, value>` entry after an item is listed for renting. 
+
+```move
+struct Listed has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Promise` struct is created for borrowing by value. The `Promise` operates as the [hot potato](../../concepts/sui-move-concepts/patterns/hot-potato.mdx) that can only be resolved by returning the item back to the extension's `Bag`.
+The Promise field lacks the `store` ability as it shouldn't be wrapped inside other objects.
+It also lacks the `drop` ability because only the `return_val` function can consume it.
+
+```move
+struct Promise {
+  item_id: Rented,
+  duration: u64,
+  start_date: u64,
+  price_per_day: u64,
+  renter: address,
+  borrower_kiosk: ID
+}
+```
+
+The `Rentable` struct is as a wrapper object that holds an asset that is being rented. Contains information relevant to the rental period, cost, and renter.
+This struct requires the `store` ability because it stores a value `T` that definitely also has `store`.
+
+```move
+struct Rentable<T: key + store> has store {
+      object: T,
+      duration: u64, // total amount of time offered for renting in days
+      start_date: Option<u64>, // initially undefined, is updated once someone rents it
+      price_per_day: u64,
+      kiosk_id: ID // the kiosk id that the object was taken from
+}
+```
+
+The `RentalPolicy` struct is a shared object that every creator mints. The struct defines the royalties the creator receives from each rent invocation.
+
+```move
+struct RentalPolicy<phantom T> has key, store {
+	id: UID,
+	balance: Balance<SUI>,
+	/// Note: Move does not support float numbers.
+	/// If you need to represent a float, you need to determine the desired precision and use a larger integer representation.
+	/// For example, percentages can be represented using basis points:
+	/// 10000 basis points represents 100% and 100 basis points represents 1%.
+	amount_bp: u64
+}
+```
+
+The `ProtectedTP` object is a shared object that creators mint to enable renting, regardless of their intent to enforce royalties. The object provides authorized access to an empty `TransferPolicy`.
+
+A protected empty transfer policy is required to facilitate the rental process so that the extension can transfer the asset without any additional rules to resolve (like lock rule, loyalty rule, and so on). If creators want to enforce royalties on rentals, they can use the `RentalPolicy` detailed previously.
+
+```move
+struct ProtectedTP<phantom T> has key, store {
+	id: UID,
+	transfer_policy: TransferPolicy<T>,
+	policy_cap: TransferPolicyCap<T>
+}
+```
+
+### Function signatures
```

Comment:
For this to be a guide that people can follow, we would also need to cover the contents of the functions, not just the function signatures, and additionally to go into why we make certain choices in the implementation etc.

Eventually, we will also want to adapt guides to have more of a narrative (so they read like a journey that the reader can follow to build up a working example step-by-step, rather than documentation to understand a system end-to-end in its finished state), but that will be a challenge for another day/PR.

Reply:
Yeah a numbered process / document would be ideal from a follow along standpoint although would require careful consideration on how the sections are split and what is needed for the tutorial (for example for these types of docs we might not need the architecture diagrams) 

Will try to include some additional explanations based on what the function body does. Would it be best if I included the body in the code block and explained above interesting lines with code comments?

Reply:
We don't have a strong guideline on this to be honest, but for now, having the code in a code block and then text before or after to explain it sounds good, thanks!

Reply:
> Eventually, we will also want to adapt guides to have more of a narrative (so they read like a journey that the reader can follow to build up a working example step-by-step, rather than documentation to understand a system end-to-end in its finished state), but that will be a challenge for another day/PR.

Do we want all examples to be in this narrative format? I'm of the opinion that both are valuable for different audiences. And because this one is not in the Examples part of the site, I was thinking it would be more of the latter.  We should discuss that.